File: Quantum Anealing/src/main/model/Graph.java
package main.model;

import java.util.List;

public class Graph {

    private final List<Vertex> vertexes;
    private final List<Edge> edges;

    public Graph(List<Vertex> vertexes, List<Edge> edges) {
        this.vertexes = vertexes;
        this.edges = edges;
    }

    public Vertex getVertexById(String id) {
        for (Vertex vertex : vertexes) {
            if (vertex.getId().equals(id)) {
                return vertex;
            }
        }
        return null;
    }

    public int getVertexIndexById(String id) {
        for (int i = 0; i < vertexes.size(); i++) {
            if (vertexes.get(i).getId().equals(id)) {
                return i;
            }
        }
        return -1;
    }

    public List<Vertex> getVertexes() {
        return vertexes;
    }

    public List<Edge> getEdges() {
        return edges;
    }
}

File: Python/Quantum Anealing/Python/model/graph.py
class Graph:
    def __init__(self, vertexes, edges):
        self.__vertexes = vertexes
        self.__edges = edges

    def get_vertexes(self):
        return self.__vertexes

    def get_edges(self):
        return self.__edges

    def get_vertex_by_id(self, id):
        if not id:
            return None

        for vertex in self.__vertexes:
            if vertex.get_id() == id:
                return vertex

        return None

    def get_vertex_index_by_id(self, id):
        if not id:
            return -1

        length = len(self.__vertexes)
        for i in range(length):
            if self.__vertexes[i] == id:
                return i

        return -1
